.github/workflows/_gen_workflows.py

```diff
@@ -37,14 +37,14 @@ class BaseTests:
         ],
     }
 
-    run_tests = {
-        "name": "Test",
+    run_checks = {
```

Comment:
I was thinking a bit about this and I don't like this approach much because if there's a single check-all which does things sequentially, you'll just get the first failure, which is not ideal.

i.e.: in the previous approach it ran the tests and then it ran the lint with a `if: always()` in the clause, so, if any of those would fail both would appear as separate failures in the ci run. With the `check-all` approach one may need one ci run to see the test failures and then another ci run to see the lint failures.

My suggestion is revert to the previous approach (ideally each separate thing that could fail which is "independent" on the others should have a separate step, so, we'd have separate step for `tests`, 'lint`, `code-format`, `typecheck` and potentially `docs`).

You could also improve the check-all to do all and report separate failures even if one fails, but I still think it's easier to see in the ci each separate failure (and one does not preclude the other -- `check-all` would be something you run locally, while the ci runs them separate).

Reply:
Agree, `check-all` something to run locally, while all the other checks ran independently in CI without a fail-fast approach, thus maximizing the CI potential of explaining to you in one go what went wrong.